server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java

```diff
@@ -33,38 +33,33 @@
 import org.sonar.db.component.ResourceDto;
 import org.sonar.db.permission.PermissionRepository;
 import org.sonar.db.permission.template.PermissionTemplateDto;
-import org.sonar.server.component.ComponentFinder;
 import org.sonar.server.permission.index.PermissionIndexer;
 import org.sonar.server.user.UserSession;
 
 import static java.util.Arrays.asList;
 import static org.sonar.server.permission.PermissionPrivilegeChecker.checkProjectAdminUserByComponentKey;
+import static org.sonar.server.ws.WsUtils.checkFoundWithOptional;
 
 @ServerSide
-public class PermissionService {
+public class PermissionTemplateService {
 
   private final DbClient dbClient;
   private final PermissionRepository permissionRepository;
   private final PermissionIndexer permissionIndexer;
   private final UserSession userSession;
-  private final ComponentFinder componentFinder;
 
-  public PermissionService(DbClient dbClient, PermissionRepository permissionRepository, PermissionIndexer permissionIndexer, UserSession userSession,
-                           ComponentFinder componentFinder) {
+  public PermissionTemplateService(DbClient dbClient, PermissionRepository permissionRepository, PermissionIndexer permissionIndexer, UserSession userSession) {
     this.dbClient = dbClient;
     this.permissionRepository = permissionRepository;
     this.permissionIndexer = permissionIndexer;
     this.userSession = userSession;
-    this.componentFinder = componentFinder;
   }
 
   /**
-   * Used by Ruby on Rails
+   * @deprecated replaced by {@link #applyDefault(DbSession, ComponentDto, Long)}, which <b>does not
+   * verify that user is authorized to administrate the component</b>.
    */
-  public List<String> globalPermissions() {
-    return GlobalPermissions.ALL;
-  }
-
+  @Deprecated
   public void applyDefaultPermissionTemplate(String componentKey) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)